cmd/nanotube/main.go

```diff
@@ -75,21 +95,41 @@ func main() {
 
 	// SIGTERM gracefully terminates with timeout
 	// SIGKILL terminates immediately
+	// SIGUSR2 triggers zero-downtime restart
 	sgn := make(chan os.Signal, 1)
-	signal.Notify(sgn, os.Interrupt, syscall.SIGTERM)
-	<-sgn
+	signal.Notify(sgn, os.Interrupt, syscall.SIGTERM, syscall.SIGUSR2)
 
-	// start termination sequence
-	close(stop)
+	for {
+		s := <-sgn
 
-	select {
-	case <-time.After(time.Second * time.Duration(cfg.TermTimeoutSec)):
-		err = lg.Sync()
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "error when syncing logger: %v", err)
+		if s == syscall.SIGUSR2 {
+			lg.Info("Reload: Got signal for reload. Checking config.")
+			_, _, _, _, _, _, err = loadBuildRegister(cfgPath, lg)
+			if err != nil {
+				lg.Error("Reload: Cannot reload: config invalid", zap.Error(err))
+				continue
+			} else {
+				lg.Info("Reload: Config OK. Starting a new instance.")
+			}
+			pid, err := n.StartProcess()
+			if err != nil {
+				lg.Error("Reload: Failed to start new process", zap.Error(err))
+			} else {
+				lg.Info("Reload: Started new process. Moved FDs.", zap.Int("pid", pid))
+			}
+			lg.Info("Termination: Staring termination sequence")
```

Comment:
Typo in "Staring" - same in line 123

Reply:
Thanks. I will upgrade my spell checker in VSCode) 
Same as the other one, it's from the other PR.